src/serializer/ResidualReactElementSerializer.js

```diff
@@ -141,17 +141,28 @@ export class ResidualReactElementSerializer {
     let dependencies = [typeValue, keyValue, refValue, propsValue, value];
     let createElement;
 
-    if (this.reactOutput === "create-element") {
+    // If we have a host component with an uppercase name, like RCTText,
+    // we cannot use JSX, so use createElement instead
```

Comment:
Please copy this comment to the other file too, it took me a bit to figure out.

If we want to be strict about outputting JSX, we could just generate an intermediate variable though.

```js
var _0 = 'RCTText';
return <_0 {...props} />
```

Reply:
^^ this approach might be nice (assuming it's not difficult) because then we don't have to complicate logic at all output code paths with a `mustVisitReactElement` argument.